Numerical/binary_search.py

```diff
@@ -0,0 +1,31 @@
+# Binary Search Algorithm
+# Input: A sorted array arr and element to be searched,key
```

Comment:
A sorted array is not necessary

Reply:
> A sorted array is not necessary

A sorted array is necessary for binary search, if the user doesn't provide it, the code should implement sort